src/wwwroot/controllers/PlanCtrl.js

```diff
@@ -57,7 +57,25 @@
       return settings.getStatusPlanInfo()
           .then(function (result) {
             vm.extraEmailsSent = 0;
-            vm.resetDate = result.data.endDate;
+            vm.isAccountClosed = result.data.accountClosed;
+            if (result.data.accountEndDate) {
+              vm.resetDate = result.data.accountEndDate;
+              if (vm.isFreeTrial){
+                vm.labelDate = 'my_plan_free_trial_end_label';
+              } else {
+                vm.labelDate = 'my_plan_account_end_label';
+              }
+            } else if (!result.data.accountEndDate) {
+              vm.labelDate = 'my_plan_renewal_date';
```

Comment:
This is what makes me uncomfortable, Does `labelDate` refer to _reset date_ or to _end date_?